### ISSUE TITLE
Bug: [explicit-function-return-type] explicit-function-return-type should provide quick fix

### ISSUE THREAD

**Issue by loynoir** [typescript-eslint/typescript-eslint#7413]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.2.0-beta&fileType=.ts&code=GYVwdgxgLglg9mABMOcBGBDATgCgJSIDeAUIohAgM5SJYCmNAvIgCwBMiGl5VUptDEFiT0%2BAXyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6RAD2MbMroDNYTMvkoB7JrWiJ8saJKKl0URNGhjokcGAC%2BIHUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```typescript
function foobar() {
  const ret = 42 as const
  return ret
}
```

### ESLint Config

```javascript
"eslintConfig": {
    "extends": [
      "standard-with-typescript",
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

After eslint fix

```ts
function foobar(): 42 {
  const ret = 42 as const
  return ret
}
```

### Actual Result

```ts
// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
function foobar() {
  const ret = 42 as const
  return ret
}
```

**Comment by bradzacher** [comment]
Duplicate of #5470 #4751 #3254 #2257.

Please use the search.

**Comment by loynoir** [comment]
@bradzacher 

Have a look of 4.

Not familiar with typescript api.

But, for example, vscode editor can show function type when hover on function id.

So, there should be a way to get types from typescript api, right?

Why about add an option `fixerUseTheReturnTypeFromTypescriptApi: false`?

If true, `@typescript-eslint/typescript-eslint` fixer suggest the 100% same type, which vscode display, when you hover on function id.

**Comment by bradzacher** [comment]
Please see this comment (linked in one of the issues)

https://github.com/typescript-eslint/typescript-eslint/issues/59#issuecomment-621942695

**Comment by loynoir** [comment]
@bradzacher 

That why I suggest `option fixerUseTheReturnTypeFromTypescriptApi: false`.

- It default to false, so there is no breaking changed.

- From name, user already accept types have a chance to not strict.

- And suggest as same as how vscode display it, so it is one and only one.

- Quote `Don't repeat yourself`. In most cases, without this feature, I found I repeat myself.

**Comment by bradzacher** [comment]
TypeScript already provides a quick action for un-annotated functions which will do exactly what you want - two clicks and it will fill in the exact return type that TS infers for the function.

![image](https://github.com/typescript-eslint/typescript-eslint/assets/7462525/b90b8249-4854-45f8-bace-0b8108d599ae)

https://github.com/typescript-eslint/typescript-eslint/assets/7462525/5b87bd69-f30b-4b6d-a7d6-c3cdd4635453

Which is another reason why there's little incentive for us to do this.

Additionally - we don't like adding conditional type information as it's very confusing for users who have a hard enough time configuring things right as it is.

**Comment by loynoir** [comment]
Ah, did not know that. 

The UI-click action is only subset of keyboard-shortcut action.

Seems like a vscode bug.
